osquery/tables/system/windows/windows_optional_features.cpp

```diff
@@ -0,0 +1,84 @@
+/**
+ *  Copyright (c) 2014-present, Facebook, Inc.
+ *  All rights reserved.
+ *
+ *  This source code is licensed under both the Apache 2.0 license (found in the
+ *  LICENSE file in the root directory of this source tree) and the GPLv2 (found
+ *  in the COPYING file in the root directory of this source tree).
+ *  You may select, at your option, one of the above-listed licenses.
+ */
+
+#include <string>
+
+#include <boost/algorithm/string.hpp>
+#include <boost/regex.hpp>
+
+#include <osquery/logger.h>
+#include <osquery/tables.h>
+
+#include "osquery/core/windows/wmi.h"
+
+namespace osquery {
+namespace tables {
+
+std::string getDismPackageFeatureStateName(uint32_t state);
```

Comment:
Why declare the function here? Can you move the definition of `getDismPackageFeatureStateName` here instead.

Reply:
When I am reading table code, the first thing I want to see is the generate function.  Typically this is because I am debugging it or trying to figure out how it works.  So I try to put helper functions at the end.